misk-hibernate/src/test/kotlin/misk/hibernate/TransacterTest.kt

```diff
@@ -374,16 +377,30 @@ class TransacterTest {
     assertThat(transacter.transaction { it.load(swid) }.name).isEqualTo("Star Wars")
   }
 
-  fun tracingAssertions() {
+  fun tracingAssertions(committed: Boolean) {
     // Assert on span, implicitly asserting that it's complete by looking at finished spans
-    assertThat(tracer.finishedSpans()).hasSize(2)
-    assertThat(tracer.finishedSpans().get(0).operationName())
+    val orderedSpans = tracer.finishedSpans().sortedBy { it.context().spanId() }
+    assertThat(orderedSpans).hasSize(4)
+
+    assertThat(orderedSpans.get(0).operationName())
+        .isEqualTo(APPLICATION_TRANSACTION_SPAN_NAME)
+    assertThat(orderedSpans.get(0).tags())
+        .containsEntry(Tags.COMPONENT.getKey(), TRANSACTER_SPAN_TAG)
+
+    assertThat(orderedSpans.get(1).operationName())
         .isEqualTo(DB_TRANSACTION_SPAN_NAME)
-    assertThat(tracer.finishedSpans().get(0).tags())
+    assertThat(orderedSpans.get(1).tags())
         .containsEntry(Tags.COMPONENT.getKey(), TRANSACTER_SPAN_TAG)
-    assertThat(tracer.finishedSpans().get(1).operationName())
-        .isEqualTo(APPLICATION_TRANSACTION_SPAN_NAME)
-    assertThat(tracer.finishedSpans().get(1).tags())
+
+    assertThat(orderedSpans.get(2).operationName())
+        .isEqualTo(DB_BEGIN_SPAN_NAME)
+    assertThat(orderedSpans.get(2).tags())
+        .containsEntry(Tags.COMPONENT.getKey(), TRANSACTER_SPAN_TAG)
+
+
+    assertThat(orderedSpans.get(3).operationName())
+        .isEqualTo(if (committed) { DB_COMMIT_SPAN_NAME } else { DB_ROLLBACK_SPAN_NAME })
```

Comment:
not a fan of conditionals inside the actual equal assertion. How about moving the conditional out, and having two assertThat's instead?

Reply:
Changed